keras/engine/topology.py

```diff
@@ -1088,7 +1088,7 @@ def add_loss(self, losses, inputs=None):
         if hasattr(self, '_losses'):
             self._losses += losses
         # Update self._per_input_updates
-        if inputs == []:
+        if inputs is []:
```

Comment:
It is not un-related... Like Nikos explained in the comments below, CNTK reserve the "==" operator for c++ side object equality, so here "input == []" will led to un-expected result, that's why I update with "input is []".....